tests/unit/s2n_security_policies_test.c

```diff
@@ -172,6 +172,21 @@ int main(int argc, char **argv)
 
     const struct s2n_security_policy *security_policy = NULL;
 
+    /* Test Deprecated Security Policies */
+    {
+        /* Ensure that every policy in the deprecated list has been removed from the supported policies list */
+        for (size_t i = 0; i < deprecrated_security_policies_len; i++) {
+            EXPECT_FAILURE_WITH_ERRNO(s2n_find_security_policy_from_version(deprecated_security_policies[i], &security_policy), S2N_ERR_DEPRECATED_SECURITY_POLICY);
+        }
+
+        /* Ensure that each policy that's been deprecated actually returns a deprecated error when requested. */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_find_security_policy_from_version("PQ-SIKE-TEST-TLS-1-0-2019-11", &security_policy), S2N_ERR_DEPRECATED_SECURITY_POLICY);
+        EXPECT_FAILURE_WITH_ERRNO(s2n_find_security_policy_from_version("PQ-SIKE-TEST-TLS-1-0-2020-02", &security_policy), S2N_ERR_DEPRECATED_SECURITY_POLICY);
+
+        /* Add list length check so that when new policies are deprecated, they are added to this test. */
+        EXPECT_EQUAL(deprecrated_security_policies_len, 2);
```

Comment:
I'm not sure about this. What do we gain from both iterating over the list, and manually repeating the list here? It seems like unnecessary work for the dev adding a new deprecated policy. Maybe just spot-check a couple policies for sanity, but don't manually check all of them?

Reply:
I think I was just being extra paranoid with these tests. Removed the length checks, and updated the comment.